pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1771,12 +1782,20 @@ static __always_inline int init_event_data(event_data_t *data, void *ctx)
     init_context(&data->context, data->task, data->config->options);
 
     data->ctx = ctx;
-    data->buf_off = sizeof(event_context_t);
     int buf_idx = SUBMIT_BUF_IDX;
     data->submit_p = bpf_map_lookup_elem(&bufs, &buf_idx);
     if (unlikely(data->submit_p == NULL))
         return 0;
 
+    submit_buf_state_t *preserved_buffer = get_preserved_submit_buffer_data();
+    if (preserved_buffer != NULL && preserved_buffer->is_initialized == true) {
+        data->buf_off = preserved_buffer->buf_off;
+        data->context.argnum = preserved_buffer->argnum;
+        delete_preserved_submit_buffer_data();
+    } else {
+        data->buf_off = sizeof(event_context_t);
+    }
+
```

Comment:
A different approach will be to save the entire event_data_t in a percpu map before calling a tail call, then getting it in the tail call and initializing the pointers to other maps in a dedicated function, not related to init_event_data